packages/core/src/useNavigationHelpers.tsx

```diff
@@ -82,4 +94,27 @@ export default function useNavigationHelpers<
     setState,
     onAction,
   ]);
+
+  const { addNavigationListener } = React.useContext(NavigationFocusedContext);
```

Comment:
I think it's better to move this code to a separate hook. `useNavigationHelpers` is to create navigation helper object, so it's not the right place for this. Maybe `useFocusedListeners`